packages/webviz-core/src/panels/Teleop/index.js

```diff
@@ -0,0 +1,291 @@
+//  found in the LICENSE file in the root directory of this source tree.
+//  You may not use this file except in compliance with the License.
+import CheckboxBlankOutlineIcon from "@mdi/svg/svg/checkbox-blank-outline.svg";
+import CheckboxMarkedIcon from "@mdi/svg/svg/checkbox-marked.svg";
+import * as React from "react";
+import { hot } from "react-hot-loader/root";
+import styled from "styled-components";
+
+import { PanelToolbarInput, PanelToolbarLabel } from "webviz-core/shared/panelToolbarStyles";
+import Autocomplete from "webviz-core/src/components/Autocomplete";
+import Button from "webviz-core/src/components/Button";
+import Flex from "webviz-core/src/components/Flex";
+import Item from "webviz-core/src/components/Menu/Item";
+import Panel from "webviz-core/src/components/Panel";
+import PanelToolbar from "webviz-core/src/components/PanelToolbar";
+import Publisher from "webviz-core/src/components/Publisher";
+import KeyListener from "webviz-core/src/components/KeyListener";
+import { PlayerCapabilities, type Topic } from "webviz-core/src/players/types";
+import colors from "webviz-core/src/styles/colors.module.scss";
+import { colors as c } from "webviz-core/src/util/sharedStyleConstants";
+import type { RosDatatypes } from "webviz-core/src/types/RosDatatypes";
+import { string } from "prop-types";
+
+type Config = {|
+  buttonColor: string,
+  topicName: string,
+|};
+
+type Props = {
+  config: Config,
+  saveConfig: ($Shape<Config>) => void,
+
+  // player state
+  capabilities: string[],
+  topics: Topic[],
+  datatypes: RosDatatypes,
+};
+
+type PanelState = {|
+  cachedProps: $Shape<Props>,
+  datatypeNames: string[],
+  parsedObject: ?any,
+  error: ?string,
+  pressing: any,
+  topic: string,
+|};
+
+const STextArea = styled.textarea`
+  width: 100%;
+  height: 100%;
+  resize: none;
+`;
+
+const STextAreaContainer = styled.div`
+  flex-grow: 1;
+  padding: 12px 0;
+`;
+
+const SErrorText = styled.div`
+  flex: 1 1 auto;
+  display: flex;
+  align-items: center;
+  padding: 4px;
+  color: ${colors.red};
+`;
+
+const SSpan = styled.span`
+  opacity: 0.8;
+`;
+const SRow = styled.div`
+  display: flex;
+  line-height: 24px;
+  flex-shrink: 0;
+`;
+
+function getTopicName(topic: Topic): string {
+  return topic.name;
+}
+
+function parseInput(value: string): $Shape<PanelState> {
+  let parsedObject;
+  let error = null;
+  try {
+    const parsedAny = JSON.parse(value);
+    if (Array.isArray(parsedAny)) {
+      error = "Message content must be an object, not an array";
+    } else if (parsedAny === null) {
+      error = "Message content must be an object, not null";
+    } else if (typeof parsedAny !== "object") {
+      error = `Message content must be an object, not ‘${typeof parsedAny}’`;
+    } else {
+      parsedObject = parsedAny;
+    }
+  } catch (e) {
+    error = value ? e.message : "";
+  }
+  return { error, parsedObject };
+}
+
+class Teleop extends React.PureComponent<Props, PanelState> {
```

Comment:
Could you convert this component to functional component. We've converted the majority of components but still a few left. Would be nice to use the new syntax.

Reply:
Done (I think so, at least, I don't really work on React and so I had to google what a functional component is). Can you confirm that I actually converted the component to functional style, please?